### ISSUE TITLE
Local Repair of Neural Networks Using Optimization. (arXiv:2109.14041v1 [cs.LG])

### ISSUE THREAD

**Issue by CoffeeKumazaki** [CoffeeKumazaki/arXiv#11531]
Local Repair of Neural Networks Using Optimization. (arXiv:2109.14041v1 [cs.LG])<br>
https://ift.tt/3ojdzNP<br>
<p>In this paper, we propose a framework to repair a pre-trained feed-forward neural network (NN) to satisfy a set of properties. We formulate the properties as a set of predicates that impose constraints on the output of NN over the target input domain. We define the NN repair problem as a Mixed Integer Quadratic Program (MIQP) to adjust the weights of a single layer subject to the given predicates while minimizing the original loss function over the original training domain. We demonstrate the application of our framework in bounding an affine transformation, correcting an erroneous NN in classification, and bounding the inputs of a NN controller.</p>
<br>
<br>
via eess.SY updates on arXiv.org http://arxiv.org/<br>
